src/vs/workbench/services/themes/common/workbenchThemeService.ts

```diff
@@ -80,12 +85,29 @@ export interface IWorkbenchThemeService extends IThemeService {
 
 }
 
+export interface IThemeScopedColorCustomizations {
+	[colorId: string]: string;
+}
+
 export interface IColorCustomizations {
-	[colorIdOrThemeSettingsId: string]: string | IColorCustomizations;
+	[colorIdOrThemeScope: string]: IThemeScopedColorCustomizations | string;
```

Comment:
I tried implementing better index types with microsoft/TypeScript#26797 and it didn't work because negated types (microsoft/TypeScript#29317) aren't implemented yet in TS. There's no valid hack for this functionality either, and to be honest it's a superfluous change.